org.eclipse.m2e.launching/src/org/eclipse/m2e/internal/launch/MavenLaunchDelegate.java

```diff
@@ -245,13 +246,15 @@ public static String readEnforcedJavaVersion(File pomDirectory, IProgressMonitor
       Optional<IContainer> container = getContainer(pomDirectory);
       if(container.isPresent()) {
         IMavenProjectRegistry projectManager = MavenPlugin.getMavenProjectRegistry();
-        IFile pomFile = container.get().getFile(Path.fromOSString(IMavenConstants.POM_FILE_NAME));
-        IMavenProjectFacade mavenProject = projectManager.create(pomFile, true, new NullProgressMonitor());
-        if(mavenProject != null) {
-          return readEnforcedVersion(mavenProject, monitor);
+        IPath pomPath = Path.fromOSString(IMavenConstants.POM_FILE_NAME);
+        if(container.get().exists(pomPath)) {
+          IFile pomFile = container.get().getFile(pomPath);
+          IMavenProjectFacade mavenProject = projectManager.create(pomFile, true, new NullProgressMonitor());
+          if(mavenProject != null) {
+            return readEnforcedVersion(mavenProject, monitor);
+          }
         }
       }
-      //TODO: handle the case if the pomDirectory points to a project not in the project. Then load the bare project.
```

Comment:
I think the comment should stay, since it still holds and covers another case than the one here.
Although the text has a mistake.